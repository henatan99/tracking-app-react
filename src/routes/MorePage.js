import React from 'react';
import { connect, useSelector } from 'react-redux';
import FooterNav from '../components/footerNav';
import More from '../containers/more';

const MorePage = () => {
  const state = useSelector((state) => state);

  return (
    <div className="measurement-page">
      <header className="measurement-page-header">
        <h3>More</h3>
      </header>
      <More username={state.user.username} userId={state.user.id} />
      <FooterNav user={state.user} />
    </div>
  );
};

const mapStateToProps = (state) => ({
  user: state.user,
});

export default connect(mapStateToProps)(MorePage);
