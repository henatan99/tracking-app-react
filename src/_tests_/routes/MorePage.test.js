import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import MorePage from '../../routes/MorePage';
import createTestStore from '../factories/storeFactory';

describe('Measureds', () => {
    let store;
    beforeEach(() => {
      store = createTestStore();
    });
    test('should match with snapshot', () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <MemoryRouter>
              <MorePage />
            </MemoryRouter>
          </Provider>,
        ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  