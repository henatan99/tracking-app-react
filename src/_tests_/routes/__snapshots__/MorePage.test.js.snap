// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Measureds should match with snapshot 1`] = `
<div
  className="measurement-page"
>
  <header
    className="measurement-page-header"
  >
    <h3>
      More
    </h3>
  </header>
  <div
    className="more"
  >
    <div
      className="user-profile"
    >
      <span
        className="user-img"
      >
        <span
          className="user-img-head"
        />
        <span
          className="user-img-body"
        />
      </span>
      <h1
        className="username"
      >
        
      </h1>
    </div>
    <ul
      className="more-ul"
    >
      <a
        className="more-link"
        href="/null/setgoal"
        onClick={[Function]}
      >
        <li
          className="more-li"
        >
          <span
            className="iconify"
            data-icon="flat-ui:goal"
            data-inline="false"
          />
          <h3
            className="more-li-title"
          >
            Your Goal
          </h3>
        </li>
      </a>
      <a
        className="more-link"
        href="/"
        onClick={[Function]}
      >
        <li
          className="more-li"
        >
          <span
            className="iconify"
            data-icon="websymbol:logout"
            data-inline="false"
          />
          <h3
            className="more-li-title"
          >
            Log Out
          </h3>
        </li>
      </a>
    </ul>
  </div>
  <nav
    className="footer-nav"
  >
    <ul
      className="footer-nav-ul"
    >
      <li
        className="footer-nav-li"
      >
        <a
          href="/undefined/measurement"
          onClick={[Function]}
        >
          <span
            className="iconify"
            data-icon="fluent-data-bar-vertical-add-24-filled"
            data-inline="false"
          />
          <span>
            Add measure
          </span>
        </a>
      </li>
      <li
        className="footer-nav-li"
      >
        <a
          href="/undefined/track/1"
          onClick={[Function]}
        >
          <span
            className="iconify"
            data-icon="iconoir:stat-up"
            data-inline="false"
          />
          <span>
            Track it
          </span>
        </a>
      </li>
      <li
        className="footer-nav-li"
      >
        <a
          href="/undefined/progress/1"
          onClick={[Function]}
        >
          <span
            className="iconify"
            data-icon="carbon:in-progress"
            data-inline="false"
          />
          <span>
            Your progress
          </span>
        </a>
      </li>
      <li
        className="footer-nav-li"
      >
        <a
          href="/undefined/more"
          onClick={[Function]}
        >
          <span
            className="iconify"
            data-icon="bi:three-dots"
            data-inline="false"
          />
          <span>
            More
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
